### PR TITLE
[ADR] Caching mailbox and email

### PR DESCRIPTION
**Pull request by hoangdat**
@chibenwa @dab246 please help me review this ADR

### CODE REVIEW

File: docs/adr/0002-caching.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Date: 2021-10-05
+
+## Status
+
+Accepted
+
+## Context
+
+- Client always make requests to back-end to refresh the data
+- User need to wait a long time to see the mailbox and email
```
Comment (chibenwa) [inline docs/adr/0002-caching.md:12]: Always requesting the same entities again and over again put a lot of pressure on the backend.

File: docs/adr/0002-caching.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- Caching the data with the [Hive](https://github.com/hivedb/hive) dependencies 
+Hive is a lightweight and blazing fast key-value database written in pure Dart. It support cross-platform: mobile, desktop and browser
+
+- `changes` method allows a client to efficiently update the state of data(Mailbox and Email)
+
+## Consequences
+
+- Minimizing the requests to the back-end 
+- Reducing the load latency in the client
```
Comment (chibenwa) [inline docs/adr/0002-caching.md:24]:
```suggestion
- Reducing the latency in the client
```

Comment (chibenwa) [reply docs/adr/0002-caching.md:24]: Also what would we need to handle offline gracefully? Pre-loading the bodies of the 10 first messages?

Comment (chibenwa) [reply docs/adr/0002-caching.md:24]: We can have a quick note here. IE in a `Follow up work`.

**Review by dab246 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**